### PR TITLE
Jetpack Connect: Introduce a Happychat button component

### PR DESCRIPTION
**Pull request by tyxla**
### Purpose

This PR introduces a Happychat button component for use in Jetpack Connect. It will be used in the footer links on most of the Jetpack Connect pages. Part of #17604 and #18149.

### How it works
It consists of a simple component that decides whether to display the Happychat button (if it's available), or whatever else we pass it as component children otherwise.

### Testing instructions

* Checkout the branch from #18397 (`update/jetpack-connect-use-happychat`), which contains this one.
* Go through some Jetpack Connect flows (see #18397 for detailed testing instructions) and verify the button looks and works properly.

### CODE REVIEW

File: client/jetpack-connect/happychat-button.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	children,
+	isChatActive,
+	isChatAvailable,
+	translate,
+} ) => {
+	if ( ! isChatAvailable && ! isChatActive ) {
+		return (
+			<div>
+				<HappychatConnection />
+				{ children }
```
[the remark this answers is not in the document]
Comment (tyxla) [reply client/jetpack-connect/happychat-button.jsx]: It's actually not a common practice, but it'll be a handy way to handle our Happychat use cases within Jetpack Connect. The idea is: either render the button, or render the children (which basically serve as a fallback in case Happychat is not available).

**Comment by tyxla** [comment]
@sirreal I understand the reasoning behind @seear's comment (https://github.com/Automattic/wp-calypso/pull/18397#issuecomment-333594558), but I'm more inclined to go with the current approach, because the main goal here is to avoid many more repetitions and help us embed Happychat into Jetpack Connect much easier. If we only extend the current Happychat button it will work well, but we'll end up having to input the same messaging, icons and logic for all of the instances within Jetpack Connect, which at this time are going to be 5 (see #18397), and will be growing as we implement it here and there.

TL;DR: I prefer using the Happychat availability selectors one more time, but preventing from a lot more repetitions as we implement Happychat throughout Jetpack Connect.

@sirreal @seear please, let me know if that sounds good.

**Comment by seear** [comment]
> @sirreal @seear please, let me know if that sounds good.

@tyxla yes, absolutely fine with that reasoning.

**Comment by tyxla** [comment]
Thank you folks 👍
